samples/subsys/zbus/remote_mock/src/mock_proxy.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2022 Rodrigo Peixoto <rodrigopex@gmail.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <stdint.h>
+#include <string.h>
+
+#include <zephyr/drivers/uart.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/net/buf.h>
+#include <zephyr/sys/crc.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/zbus/zbus.h>
+LOG_MODULE_DECLARE(zbus, CONFIG_ZBUS_LOG_LEVEL);
+
+#include "zbus_messages.h"
+const struct device *uart_dev = DEVICE_DT_GET(DT_NODELABEL(uart1));
+
+ZBUS_CHAN_DECLARE(sensor_data);
+
+ZBUS_CHAN_DEFINE(version,	 /* Name */
+		 true,		 /* Read only */
+		 struct version, /* Message type */
+
+		 NULL,		       /* Validator */
+		 ZBUS_OBSERVERS_EMPTY, /* observers */
+		 ZBUS_MSG_INIT(.major = 0, .minor = 1,
+			       .build = 1023) /* Initial value major 0, minor 1, build 1023 */
+);
+
+ZBUS_CHAN_DEFINE(start_measurement, /* Name */
+		 false,		    /* Read only */
+		 struct action,	    /* Message type */
+
+		 NULL,				    /* Validator */
+		 ZBUS_OBSERVERS(proxy, peripheral), /* observers */
+		 ZBUS_MSG_INIT(false)		    /* Initial value */
+);
+
+uint8_t encoder(struct zbus_channel *chan)
+{
+	if (chan == &sensor_data) {
+		return 1;
+	} else if (chan == &start_measurement) {
+		return 2;
+	}
+	return 0;
+}
+
+struct zbus_channel *decoder(uint8_t chan_idx)
+{
+	if (chan_idx == 1) {
+		return &sensor_data;
+	} else if (chan_idx == 2) {
+		return &start_measurement;
+	}
+	return NULL;
+}
+
+void proxy_callback(struct zbus_channel *chan)
+{
+	bool *generated_by_the_bridge = (bool *)chan->user_data;
+
+	if (*generated_by_the_bridge) {
+		*generated_by_the_bridge = false;
+		LOG_DBG("discard loopback event (channel %s)", chan->name);
+	} else {
+		uart_poll_out(uart_dev, '$');
+		uart_poll_out(uart_dev, encoder(chan));
+		for (int i = 0; i < chan->message_size; ++i) {
+			uart_poll_out(uart_dev, ((unsigned char *)chan->message)[i]);
+		}
+		uart_poll_out(uart_dev, '*');
+		LOG_DBG("sending message to host (channel "
+			"%s)",
+			chan->name);
+	}
+}
+
+ZBUS_LISTENER_DEFINE(proxy, proxy_callback);
+
+void main(void)
+{
+	LOG_DBG("[Mock Proxy] Started.");
+}
+
+void mock_proxy_rx_thread(void)
+{
+	LOG_DBG("[Mock Proxy RX] Started.");
+
+	struct net_buf_simple *rx_buf = NET_BUF_SIMPLE(64);
+
+	net_buf_simple_init(rx_buf, 0);
+
+	uint8_t byte = 0;
+
+	while (1) {
+		while (uart_poll_in(uart_dev, &byte) < 0) {
+			/* Allow other thread/workqueue to work. */
+			k_msleep(50);
+		}
+		if (byte == '*') {
+			if (rx_buf->len <= 1) {
+				LOG_DBG("[Mock Proxy RX] Discard invalid sequence");
+				/* '*' indicates the end of a sentence. Sometimes it is
+				 * necessary to flush more than on ensure sending it from
+				 * the python script. The code must discard when there is no
+				 * other data at the buffer.
+				 */
+			} else {
+				if ('$' == net_buf_simple_pull_u8(rx_buf)) {
```

Comment:
This is starting to get harsh with the indentation, how about moving this block in a separate function? (either from here or from line 104)